packages/flutter_driver/lib/src/retry.dart

```diff
@@ -28,20 +32,26 @@ Future<dynamic> retry(Action action, Duration timeout,
   while(!success && sw.elapsed < timeout) {
     try {
       result = await action();
-      success = true;
+      if (predicate == null || predicate(result) == true) {
```

Comment:
What does it mean for it to return null? Is it an error?

Unless "null" means something different than "false", or "null" is for some reason easier to return than "false" in important cases, I'd just define the API as needing to return true or false. That's what documentation is for. :-)